### ISSUE TITLE
Should all the constant-only interfaces be namespaces instead?

### ISSUE THREAD

**Issue by bzbarsky** [gpuweb/gpuweb#304]
It's not clear to me that there will ever be `WebGPUBufferUsage` objects, say.  And if there will not, if this is just a place to put some constants, it should probably be a namespace, not an interface.

**Comment by bzbarsky** [comment]
Oh, and this applies to a bunch of interfaces: `WebGPUTextureDimension`, `WebGPUTextureFormat`, etc, etc.

**Comment by kainino0x** [comment]
This makes sense. But I just tried it and there is no IDL syntax for constants inside namespaces. The only option is `readonly attribute`, but that doesn't allow us to do `readonly attribute u32 NONE = 0;`, for example. Am I missing something?

**Comment by bzbarsky** [comment]
You're not missing anything.  I forgot that https://github.com/heycam/webidl/issues/440 was resolved as "let's not until there's a use case".  I just reopened it.

**Comment by domenic** [comment]
I'd like to +1 this issue. I'm happy to help resolve https://github.com/heycam/webidl/issues/440. It'd be great if there were no `GPUBufferUsage.prototype` or `GPUBufferUsage.prototype.constructor` objects.

**Comment by domenic** [comment]
heycam/webidl#440 is resolved, and you can now use constants on namespaces!

**Comment by toji** [comment]
I'll make the update to the spec IDL. Thanks!

**Comment by Kangz** [comment]
It's already done in https://github.com/gpuweb/gpuweb/pull/1709 I believe.

**Comment by toji** [comment]
So it is! Thanks!

**Comment by ExE-Boss** [comment]
@Kangz <https://github.com/gpuweb/gpuweb/pull/1709> hasn’t been merged yet though.

**Comment by Kangz** [comment]
Thanks, reopening until the PR is landaded.
